gateway/auth.py

```diff
@@ -100,14 +101,27 @@ def wrapper(*args, **kwargs):
             user = F8aUser(decoded.get('email', 'nobody@nowhere.nodomain'))
 
         if user:
-            g.current_user = user
+            if user_whitelisted(user):
+                g.current_user = user
+            else:
+                g.current_user = F8aUser('unauthenticated@no.auth.token')
+                raise HTTPError(401, 'User needs to be whitelisted')
         else:
             g.current_user = F8aUser('unauthenticated@no.auth.token')
             raise HTTPError(401, 'Authentication required')
         return view(*args, **kwargs)
 
     return wrapper
 
+def user_whitelisted(user,users_whitelist='users_whitelist'):
+
+    with open(users_whitelist, 'r') as f:
+        white_list = f.readlines()
+
+    if user in white_list:
+        return True
+    else:
+        return False
```

Comment:
shorter form? `return user in whitelist`

Reply:
thanks.